### PR TITLE
return node is starting for all api calls when needed

### PR DESCRIPTION
**Pull request by bogdan-rosianu**
refactored the way `node is starting` error is returned. before this PR, there were some endpoints that didn't return an error and also returned code 200

### CODE REVIEW

File: api/groups/nodeGroup_test.go [hunk 1]
```diff
@@ -125,6 +126,66 @@ func TestHeartbeatstatus(t *testing.T) {
 	assert.NotEqual(t, "", statusRsp.Message)
 }
 
+func TestP2PMetrics_ShouldReturnErrorIfFacadeReturnsError(t *testing.T) {
```
Comment (sstanculeanu) [inline api/groups/nodeGroup_test.go:129]: perhaps a new `TestQueryDebug_ShouldReturnErrorIfFacadeReturnsError` for extra coverage

File: facade/initial/disabledStatusMetrics.go [hunk 2]
```diff
@@ -13,44 +12,42 @@ func NewDisabledStatusMetricsHandler() *disabledStatusMetricsHandler {
...
+func (d *disabledStatusMetricsHandler) NetworkMetrics() (map[string]interface{}, error) {
+	return getReturnValues()
 }
 
-func getReturnMap() map[string]interface{} {
-	return map[string]interface{}{
-		responseKey: responseValue,
-	}
+func getReturnValues() (map[string]interface{}, error) {
+	return map[string]interface{}{}, errNodeStarting
```
Comment (sstanculeanu) [inline facade/initial/disabledStatusMetrics.go:50]:
```suggestion
	return map[string]interface{}{
		responseKey: responseValue,
	}, errNodeStarting
```
or remove responseValue as it is not used anymore

File: integrationTests/mock/statusMetricsStub.go [hunk 3]
```diff
@@ -2,54 +2,54 @@ package mock
 
 // StatusMetricsStub -
 type StatusMetricsStub struct {
```
Comment (ssd04) [inline integrationTests/mock/statusMetricsStub.go:4]: i think we can remove also this mock and reference the newly created one from testscommon

File: api/groups/nodeGroup_test.go (hunk 1 above)
Comment (bogdan-rosianu) [reply api/groups/nodeGroup_test.go:129]: the situation is already covered in `TestQueryDebug_GetQueryErrorsShouldErr`

File: facade/initial/disabledStatusMetrics.go (hunk 2 above)
Comment (bogdan-rosianu) [reply facade/initial/disabledStatusMetrics.go:50]: removed

File: integrationTests/mock/statusMetricsStub.go (hunk 3 above)
Comment (bogdan-rosianu) [reply integrationTests/mock/statusMetricsStub.go]: good catch. done

**Review by sstanculeanu — DISMISSED** [review]

**Review by ssd04 — APPROVED** [review]

**Review by sstanculeanu — APPROVED** [review]

**Review by gabi-vuls — APPROVED** [review]
System test passed.

**Pull request merged**